Library/Homebrew/rubocops/lines.rb

```diff
@@ -195,6 +195,48 @@ def autocorrect(node)
         end
       end
 
+      class ShellCmd < FormulaCop
+        def audit_formula(_node, _class_node, _parent_class_node, body_node)
+          popen_commands = [
+            :popen,
+            :popen_read,
+            :safe_popen_read,
+            :popen_write,
+            :safe_popen_write,
+          ]
+
+          test = find_block(body_node, :test)
+          unless test.nil?
+            popen_commands.each do |command|
+              find_instance_method_call(test, "Utils", command) do
+                problem "Use `shell_command` instead of `Utils.#{command}` in the `test` block"
+              end
+            end
+          end
+
+          [:popen_read, :popen_write].each do |unsafe_command|
+            find_instance_method_call(body_node, "Utils", unsafe_command) do |method|
+              unless method.method_name == :test
+                problem "Use `Utils.safe_#{unsafe_command}` instead of `Utils.#{unsafe_command}`"
+              end
+            end
+          end
+        end
+
+        def autocorrect(node)
+          lambda do |corrector|
+            test = find_block(@body, :test)
+            unless test.nil?
+              if method_called_in_block?(test, node.method_name)
+                corrector.replace(node.loc.selector, "shell_command")
+                return
+              end
+            end
+            corrector.replace(node.loc.selector, "safe_#{node.method_name}")
```

Comment:
This doesn't work correctly.

What I want to put is:

```ruby
if node is in the test block
  corrector.replace(node.loc.selector, "shell_command")
else
  corrector.replace(node.loc.selector, "safe_#{node.method_name}")
end
```

I can't find a method that works correctly for that. This fails because it will correct `Utils.popen_read` to `shell_command` if a `test` block exists and ANY `Utils.popen_read` function is in that `test` block.

This means that

```ruby
def install
  Utils.popen_read "foo"
end

test do
  Utils.popen_read "foo"
end
```

will become

```ruby
def install
  shell_command "foo"
end

test do
  shell_command "foo"
end
```

instead of

```ruby
def install
  Utils.safe_popen_read "foo"
end

test do
  shell_command "foo"
end
```